signbank/dictionary/csv_import.py

```diff
@@ -259,274 +271,277 @@ def confirm_import_nzsl_share_gloss_csv(request):
-                ))
+                    language=language_mi,
+                    translations=maori_words[0]
+                )
+                if len(maori_words) > 1:
+                    translation.translations_secondary = ", ".join(maori_words[1:])
+
+                translations.append(translation)
+
+            # Prepare new idgloss and signer fields for bulk update
+            gloss.idgloss = f"{word_en}:{gloss.pk}"
+            gloss.signer = signer_dict[gloss_data["contributor_username"]]
+            bulk_update_glosses.append(gloss)
+
+            # Create comment for gloss_data notes
+            comments.append(Comment(
+                content_type=gloss_content_type,
+                object_pk=gloss.pk,
+                user_name=gloss_data.get("contributor_username", ""),
+                comment=gloss_data.get("notes", ""),
+                site=site,
+                is_public=False,
+                submit_date=comment_submit_date
+            ))
+            if gloss_data.get("sign_comments", None):
+                # create Comments for all gloss_data sign_comments
+                for comment in gloss_data["sign_comments"].split("|"):
+                    try:
+                        comment_content = comment.split(":")
+                        user_name = comment_content[0]
+                        comment_content = comment_content[1]
+                    except IndexError:
+                        comment_content = comment
+                        user_name = "Unknown"
+                    comments.append(Comment(
+                        content_type=gloss_content_type,
+                        object_pk=gloss.pk,
+                        user_name=user_name,
+                        comment=comment_content,
+                        site=site,
+                        is_public=False,
+                        submit_date=comment_submit_date
+                    ))
+
+            # Add ShareValidationAggregation
+            bulk_share_validation_aggregations.append(ShareValidationAggregation(
+                gloss=gloss,
+                agrees=int(gloss_data["agrees"]),
+                disagrees=int(gloss_data["disagrees"])
+            ))
 
-                # prep videos, illustrations and usage example for video retrieval
-                if gloss_data.get("videos", None):
-                    video_url = gloss_data["videos"]
+            # prep videos, illustrations and usage example for video retrieval
+            if gloss_data.get("videos", None):
+                video_url = gloss_data["videos"]
+                extension = video_url[-3:]
+                file_name = (
+                    f"{gloss.pk}-{word_en}.{gloss.pk}_video.{extension}"
+                )
+
+                glossvideo = {
+                    "url": video_url,
+                    "file_name": file_name,
+                    "gloss_pk": gloss.pk,
+                    "title": "Main",
```

Comment:
**question**: am I getting mixed up here or should this be setting the video type?

Reply:
needs a rebase 😉 so yes.